contracts/zero-ex/contracts/src/features/UniswapV2ForkFeature.sol

```diff
@@ -0,0 +1,200 @@
+
+        if (unwrapWeth) {
+            IEtherTokenV06(WETH).withdraw(buyAmount);
+            (bool sent, ) = msg.sender.call{value: buyAmount}("");
+            require(sent, "UniswapV2ForkFeature/FAILED_TO_SEND_ETHER");
+        }
+    }
+
+    function _sortTokens(
+        IERC20TokenV06 tokenA,
+        IERC20TokenV06 tokenB
+    )
+        internal
+        pure
+        returns (IERC20TokenV06 token0, IERC20TokenV06 token1)
+    {
+        require(tokenA != tokenB, 'UniswapV2ForkFeature/IDENTICAL_ADDRESSES');
+        (token0, token1) = tokenA < tokenB ? (tokenA, tokenB) : (tokenB, tokenA);
+        require(token0 != IERC20TokenV06(0), 'UniswapV2ForkFeature/ZERO_ADDRESS');
+    }
+
+    function _getAmountOut(
+        uint amountIn,
+        uint reserveIn,
+        uint reserveOut
+    )
+        internal
+        pure
+        returns (uint amountOut)
+    {
+        require(amountIn > 0, 'UniswapV2ForkFeature/INSUFFICIENT_INPUT_AMOUNT');
+        require(reserveIn > 0 && reserveOut > 0, 'UniswapV2ForkFeature/INSUFFICIENT_LIQUIDITY');
+        uint amountInWithFee = amountIn.safeMul(997);
+        uint numerator = amountInWithFee.safeMul(reserveOut);
+        uint denominator = reserveIn.safeMul(1000).safeMul(amountInWithFee);
+        amountOut = numerator / denominator;
+    }
+
+    function _pairFor(
+        IERC20TokenV06 tokenA,
+        IERC20TokenV06 tokenB
+    )
+        internal
+        view
+        returns (IUniswapV2ForkPair)
+    {
+        (IERC20TokenV06 token0, IERC20TokenV06 token1) = _sortTokens(tokenA, tokenB);
+        return IUniswapV2ForkPair(address(uint(keccak256(abi.encodePacked(
+            hex'ff',
+            UNI_FACTORY_ADDRESS,
+            keccak256(abi.encodePacked(address(token0), address(token1))),
+            UNI_POOL_INIT_CODE_HASH
+        )))));
+    }
+
+    function _swap(
+        IERC20TokenV06[] memory tokens,
+        uint256 sellAmount,
+        address _to
+    )
+        internal
+        returns (uint[] memory amounts)
+    {
+        require(tokens.length >= 2, 'UniswapV2ForkFeature/INVALID_PATH');
+
+        amounts = new uint[](tokens.length);
+        amounts[0] = sellAmount;
+
+        for (uint i; i < tokens.length - 1; i++) {
+            (IERC20TokenV06 tokenA, IERC20TokenV06 tokenB, IERC20TokenV06 tokenC) =
+                (tokens[i], tokens[i + 1], tokens[i + 2]);
+
+            address to = i < tokens.length - 2 ? address(_pairFor(tokenB, tokenC)) : _to;
+            amounts[i + 1] = _swapInternal(amounts[i], tokenA, tokenB, to);
+        }
+    }
+
+    function _swapInternal(
+        uint256 amountIn,
+        IERC20TokenV06 tokenA,
+        IERC20TokenV06 tokenB,
+        address to
+    )
+        internal
+        returns (uint amountOut)
+    {
+        IUniswapV2ForkPair pairCurrent = _pairFor(tokenA, tokenB);
+
+        (IERC20TokenV06 token0,) = _sortTokens(tokenA, tokenB);
```

Comment:
could just check `tokenA < tokenB` instead of `tokenA == token0` for less gas

Reply:
I think they are the same? https://github.com/crytic/evm-opcodes

Reply:
I mean if you just replace the `tokenA == token0` with `tokenA < tokenB` you don't have to call `_sortTokens()` to compute `token0` at all.